### PR TITLE
Feature/service cleanup

### PR DESCRIPTION
**Pull request by whoswendy**
This PR includes changes to clean up code in service modules.
Changes include
- changes to controller exception handling
- javadocs
- readmes

### CODE REVIEW

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/MappedDiagnosticContextRequestFieldSetter.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    public static final String REQUEST_MDC_KEY = "request";
+
+    private final RequestPayloadConverter requestPayloadConverter;
+
+    public MappedDiagnosticContextRequestFieldSetter(RequestPayloadConverter requestPayloadConverter) {
+        this.requestPayloadConverter = requestPayloadConverter;
+    }
+
+    /**
+     * Set.
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/MappedDiagnosticContextRequestFieldSetter.java]: Be more specific

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/MappedDiagnosticContextRequestFieldSetter.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    /**
+     * The constant CORRELATION_IDENTIFIER_MDC_KEY.
+     */
+    public static final String CORRELATION_IDENTIFIER_MDC_KEY = "correlationIdentifier";
+
+    /**
+     * The constant REQUEST_MDC_KEY.
+     */
+    public static final String REQUEST_MDC_KEY = "request";
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/MappedDiagnosticContextRequestFieldSetter.java]: is it needed

File: service/synapse-service-graphql/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# synapse-service-graphql
+
+## Description
+
```
Comment (guttff) [inline service/synapse-service-graphql/README.md:4]: add description

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/InternalServerErrorResponseCreator.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ * {@code InternalServerErrorResponseCreator} handles internal server errors
+ */
+@Component
+public class InternalServerErrorResponseCreator {
+
+    private final ErrorMessagePropertyReader messagePropertyReader;
+
+    private final MappedDiagnosticContextRequestFieldSetter mappedDiagnosticContextRequestFieldSetter;
+
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/InternalServerErrorResponseCreator.java]: javadoc

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/MDCKeys.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ * the License.
+ */
+package io.americanexpress.synapse.service.rest.controller.exceptionhandler;
+
+/**
+ * {@code MDCKeys} contains constants for MDC keys.
+ */
+class MDCKeys {
+
+    private MDCKeys() {
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/MDCKeys.java:24]: javadoc

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/MappedDiagnosticContextRequestFieldSetter.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+ */
+@Component
+public class MappedDiagnosticContextRequestFieldSetter {
+
+    /**
+     * Used to log the exceptions
+     */
+    private final XLogger logger = XLoggerFactory.getXLogger(getClass());
+
+    private final RequestPayloadConverter requestPayloadConverter;
```
Comment (guttff) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/MappedDiagnosticContextRequestFieldSetter.java:39]: javadoc

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/MDCKeys.java (hunk 5 above)
Comment (whoswendy) [reply service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/MDCKeys.java:24]: done

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/MappedDiagnosticContextRequestFieldSetter.java (hunk 6 above)
Comment (whoswendy) [reply service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/MappedDiagnosticContextRequestFieldSetter.java:39]: done

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/InternalServerErrorResponseCreator.java (hunk 4 above)
Comment (whoswendy) [reply service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/InternalServerErrorResponseCreator.java]: done

File: service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/ControllerExceptionHandler.java [hunk 7]
```diff
@@ -157,51 +178,81 @@ public ResponseEntity<ErrorResponse> handleBindException(BindException bindExcep
...
+                "The user has exited prior to service completion.",
+                "ClientAbortException was caught.");
 
-        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(errorResponse);
+        // Note: there is currently no industry HTTP standard status code for client abort exception
+        //       i.e. although the client chose to close the connection implies a 4XX series,
+        //       there is no existing, matching 4XX status code of this description,
+        //       so we are returning custom 499 instead
+        // Considerations: 408 REQUEST TIMEOUT (the request did not timeout but rather, the response is waiting)
+        return ResponseEntity.status(499).body(errorResponse);
```
Comment (andCastStatic) [inline service/synapse-service-rest/src/main/java/io/americanexpress/synapse/service/rest/controller/exceptionhandler/ControllerExceptionHandler.java:255]: interesting

**Review by andCastStatic — APPROVED** [review]

**Pull request closed without merging**